zookeeper-server/src/main/java/org/apache/zookeeper/cli/SyncCommand.java

```diff
@@ -74,7 +75,7 @@ public void processResult(int rc, String path, Object ctx) {
             if (resultCode == 0) {
                 out.println("Sync is OK");
             } else {
-                out.println("Sync has failed. rc=" + resultCode);
+                throw new CliWrapperException(new KeeperException.NoNodeException(path));
             }
```

Comment:
- You may never get a `rc` code which tell you whether the path is exist because the server doesn't send it to the client
- One alternative way is `getData` or `exist`  to check the existence of path after `sync`
- `sync` api has problems and not user-friendly, we will depreciate it in the further with a new linearity read api

Reply:
Hi @maoling if I understand the alternative that you mentioned above correctly then I should use zookeeper.exists(path, false) after zookeeper.sync. Is this correct or have I misinterpreted your thought??

Reply:
Yes, it's what I thought

Reply:
Ok @maoling I will use zookeeper.exists(path, false) after you get the result code and add a condition to check if stat!=null && resultcode == 0 then sync is ok then throw the exception.

Also will add a unit test for failure

Reply:
@maoling I ended up using zookeeper.getData instead of exists in my latest commit and when path doesn't exist NoNode Exception is thrown which we can find in javadoc of getData so only change was to catch the exception and I have also changed the unit test accordingly.